dagshub/streaming/filesystem.py

```diff
@@ -246,7 +246,8 @@ def _special_file(self):
         # TODO Include more information in this file
         return b'v0\n'
 
-    def open(self, file: Union[bytes, PathLike, int], mode: str = 'r', *args, opener=None, **kwargs):
+    def open(self, file: Union[bytes, PathLike, int], mode: str = 'r', buffering=-1, encoding=None,
```

Comment:
I think you need to get rid of the type hints in this and the other hook functions, to ensure compatibility.
Otherwise I don't see any issue but I still need to understand the code better